vignettes/Styling_KM_plots.Rmd

```diff
@@ -0,0 +1,136 @@
+lung_cohort <- survival::lung
+
+lung_cohort <- lung_cohort %>%  
+  dplyr::mutate(sex = as.factor(ifelse(sex == 1, "Male", "Female")))  %>%  
+  dplyr::mutate(status = status -1) %>%
+  dplyr::rename(Age = "age", Sex = "sex", Status = "status", Days = "time") 
+
+lung_suvival_object <- lung_cohort %>%
+  visR::estimate_KM(strata = "Sex", CNSR = "Status", AVAL = "Days")
+```
+
+### Plotting the generated `survfit` object without adjustments
+```{r default-ggplot2-plot}
+p <- lung_suvival_object %>%
+  visR::visr()
+p
+```
+
+As we can, the plot shows the default [`ggplot2::theme_grey()`](https://ggplot2.tidyverse.org/reference/ggtheme.html) style with a grey background, a visible grid and the default `ggplot2` colours.
+
+## Using `ggplot2` to style the plot
+
+Since `visR::visr()` also generates a valid `ggplot` object as an output, we can use the conventional styling logic and options that `ggplot2` provides, as shown below. 
+
+```{r styling-with-ggplot2}
+p +
+  ggplot2::theme_bw() +
+  ggplot2::theme(legend.position = "top") +
+  ggplot2::scale_color_manual(values = c("red", "blue"))
+```
+
+However, `visR` also provides functions to adjust common aesthetics more easily and with less code. 
+
+## Using `visR` to style the plot
+
+The most direct option to style plots generated through `visR::visr()` is by using the parameters that the function provides. Internally, parameters like the y-axis label are automatically deducted from the used function. The following example demonstrates the options exposed.
+
+```{r visr-parameter-styling}
+lung_suvival_object %>%
+  visR::visr(x_label = "Time",
+             y_label = NULL, # NULL (default) causes the label to be deducted from the used function
+             x_units = "days",
+             x_ticks = seq(0, 1200, 200),
+             y_ticks = seq(0, 100, 20),
+             fun = "pct",
+             legend_position = "top") 
+```
+However, these rather minimal adjustments usually don't cover all the things a user wants to modify. Therefore, we provide two additional function to adjust more aesthetics: `visR::define_theme()` and `visR::apply_theme()`. The first one provides an easy wrapper to create a nested list of list with styling options that is then applied to the plot by the second function.
+
+### Defining a `visR_theme` using `visR::define_theme()`
+If no further options are provided to `visR::define_theme()`, it nonetheless returns a very minimal list of reasonable defaults.
+```{r visr-define_theme-empty}
+visR::define_theme() 
+```
+However, this function also takes several other styling options. The currently usable ones are displayed below. One particular use that we had in mind when writing this function was, that we wanted to have the option to define the different colours for the strata once and then to not have to worry about all of them being present. When defining the strata through a list of colours, the absence of data for a particular option would shift the order. 
```

Comment:
If the strata are factors as they are expected, then no shift would happen. I would rephrase this a bit.